druid/src/contexts.rs

```diff
@@ -242,6 +276,66 @@ impl<'a> EventCtx<'a> {
         &self.window
     }
 
+    /// Create a new window.
+    /// `T` must represent the application state provided to [`AppLauncher::launch`].
+    ///
+    /// [`AppLauncher::launch`]: struct.AppLauncher.html#method.launch
+    pub fn new_window<T: Any>(&mut self, desc: WindowDesc<T>) {
+        if self.state_types.check_window_desc::<T>() {
+            self.submit_command(
+                Command::one_shot(commands::NEW_WINDOW, desc),
+                Target::Global,
+            );
+        } else {
+            const MSG: &str = "WindowDesc<T> - T must match the application state.";
+            if cfg!(debug_assertions) {
+                panic!(MSG);
+            } else {
+                log::error!("EventCtx::new_window: {}", MSG)
+            }
+        }
+    }
+
+    /// Set the menu of the window containing the event handling widget.
+    /// `T` must represent the application state provided to [`AppLauncher::launch`].
+    ///
+    /// [`AppLauncher::launch`]: struct.AppLauncher.html#method.launch
+    pub fn set_menu<T: Any>(&mut self, menu: MenuDesc<T>) {
+        if self.state_types.check_menu_desc::<T>() {
+            self.submit_command(
+                Command::new(commands::SET_MENU, menu),
+                Target::Window(self.window_id),
+            );
+        } else {
+            const MSG: &str = "MenuDesc<T> - T must match the application state.";
+            if cfg!(debug_assertions) {
+                panic!(MSG);
+            } else {
+                log::error!("EventCtx::set_menu: {}", MSG)
+            }
+        }
+    }
+
+    /// Show the context menu in the window containing the event handling widget.
+    /// `T` must represent the application state provided to [`AppLauncher::launch`].
+    ///
+    /// [`AppLauncher::launch`]: struct.AppLauncher.html#method.launch
+    pub fn show_context_menu<T: Any>(&mut self, menu: ContextMenu<T>) {
+        if self.state_types.check_context_menu::<T>() {
+            self.submit_command(
+                Command::new(commands::SHOW_CONTEXT_MENU, menu),
+                Target::Window(self.window_id),
+            );
+        } else {
+            const MSG: &str = "ContextMenu<T> - T must match the application state.";
+            if cfg!(debug_assertions) {
```

Comment:
if this becomes a pattern we could consider a custom macro, like `log_or_panic!(...)`.

Reply:
I would like it if this becomes a pattern (or `log_or_panic!`) as `panic!` makes debugging easier due to the back trace while `log::error!` does not crash the app once you ship it.